src/ResourceManager/Network/Commands.Network/Models/PSPrivateIpAddressAvailabilityResponse.cs

```diff
@@ -0,0 +1,25 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Commands.Network.Models
+{
+    public class PSPrivateIpAddressAvailabilityResponse
+    {
+        public bool Available { get; set; }
+
+        public List<string> AvailableIpAddresses { get; set; }
```

Comment:
you would need to add a serialization function like AvailableIpAddressesText. Check how the List<PSSubnet> is modelled in PSVirtualNetwork. 

Also rename to AvailableIPAddresses.

Also you would need to add this class to format.xml